### PR TITLE
Update sulfur CI scripts

### LINKED ISSUE #4561: Build fail with cuda 12.1 and ICX23
**Describe the bug**

Build failure message on intended CI configuration on sulfur using AFQMC + Offload:

```[685/899] Building CUDA object src/AFQMC/CMakeFiles/afqmc.dir/Numerics/detail/CUDA/Kernels/sum.cu.o
FAILED: src/AFQMC/CMakeFiles/afqmc.dir/Numerics/detail/CUDA/Kernels/sum.cu.o 
/usr/local/cuda-12.1/bin/nvcc -forward-unknown-to-host-compiler -ccbin=/usr/lib64/openmpi/bin/mpicxx -DADD_ -DH5_USE_110_API -DHAVE_CONFIG_H -DHAVE_LIBBOOST -DHAVE_MKL -DHAVE_MKL_VML -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX -DOPENMP_NO_COMPLEX -DUSE_MPI -D_MPICC_H -Drestrict=__restrict__ -I/home/wfg/workspace/qmcpack/build/src -I/home/wfg/workspace/qmcpack/src -I/home/wfg/workspace/qmcpack/src/Utilities -I/home/wfg/workspace/qmcpack/src/Platforms/Host -I/home/wfg/workspace/qmcpack/external_codes/mpi_wrapper/mpi3/include -I/home/wfg/workspace/qmcpack/src/io -I/home/wfg/workspace/qmcpack/src/Containers -I/home/wfg/workspace/qmcpack/src/Platforms -I/home/wfg/workspace/qmcpack/src/io/OhmmsData/.. -I/home/wfg/workspace/qmcpack/external_codes/boost_multi/multi/include -isystem /usr/local/cuda-12.1/include -isystem /usr/include/libxml2 -std=c++17 --allow-unsupported-compiler -Drestrict=__restrict__ -DNO_CUDA_MAIN -O2 -g -DNDEBUG -std=c++17 --generate-code=arch=compute_70,code=[compute_70,sm_70] -MD -MT src/AFQMC/CMakeFiles/afqmc.dir/Numerics/detail/CUDA/Kernels/sum.cu.o -MF src/AFQMC/CMakeFiles/afqmc.dir/Numerics/detail/CUDA/Kernels/sum.cu.o.d -x cu -c /home/wfg/workspace/qmcpack/src/AFQMC/Numerics/detail/CUDA/Kernels/sum.cu -o src/AFQMC/CMakeFiles/afqmc.dir/Numerics/detail/CUDA/Kernels/sum.cu.o
/usr/local/cuda-12.1/include/thrust/detail/numeric_traits.h(35): error: integral_constant is not a template
      : integral_constant<bool, std::numeric_limits<Number>::is_signed>
        ^
```

**To Reproduce**
1. develop branch 
2. cmake command: 
   Source first Intel oneAPI env:
   `source /opt/intel/oneapi/setvars.sh`
   
```
     cmake -GNinja \
              -DCMAKE_C_COMPILER=/usr/lib64/openmpi/bin/mpicc \
              -DCMAKE_CXX_COMPILER=/usr/lib64/openmpi/bin/mpicxx \
              -DMPIEXEC_EXECUTABLE=/usr/lib64/openmpi/bin/mpirun \
              -DBUILD_AFQMC=ON \
              -DENABLE_CUDA=ON \
              -DQMC_GPU_ARCHS=sm_70 \
              -DQMC_COMPLEX=0 \
              -DQMC_MIXED_PRECISION=0 \
              -DCMAKE_BUILD_TYPE=RelWithDebInfo \
              -DQMC_DATA=$QMC_DATA_DIR \
              {QMCPACK_SRC_DIR}
```
  
4. full program/test invocation command : `ninja`
5. additional steps

**Expected behavior**
Should build and tests pass. 

**System:**
 - system name: sulfur
 - modules loaded: openAPI env (see above)
 - other systems where this is reproducible: didn't test on other systems

**Additional context**
It seems like a issue with ICX and C++17.

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes
Current sulfur CI is down after major OS upgrade
Refactor sulfur CI sulfur-1 and sulfur-2 CI jobs
Moves from Intel classic compiler `icc` to oneAPI `icx`
Related to #4536

WIP, needs:
- [ ] resolve #4561 
- [x] cuda 11.2 on sulfur

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
sulfur, WIP

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
You can install 11.2 toolkit via spack in $HOME to workaround the bugs (?) for offload builds. This is working in the nightlies just fine with the 12.1p1 driver - I can push the scripts if it will help.

**Comment by williamfgc** [comment]
ornl sulfur CI runners are back. This needs to be merged to test.

**Comment by markdewing** [comment]
What's the difference between run_step_ornl-sulfur-1.sh and run_step_ornl-sulfur-2.sh ?
It looks like *-1.sh is for the CPU build, and *-2.sh is for the offload build?

File: .github/workflows/ci-github-actions-self-hosted.yaml [hunk 1]
```diff
@@ -115,13 +115,13 @@ jobs:
...
       matrix:
         jobname: [
-            Clang15-MPI-CUDA-AFQMC-Offload-Real-Mixed, # auxiliary field, offload
-            Clang15-MPI-CUDA-AFQMC-Offload-Real,
-            Clang15-MPI-CUDA-AFQMC-Offload-Complex-Mixed,
-            Clang15-MPI-CUDA-AFQMC-Offload-Complex,
-            Intel21-MPI-CUDA-AFQMC-Real-Mixed, # auxiliary field, requires MPI
-            Intel21-MPI-CUDA-AFQMC-Complex-Mixed,
-            Intel21-MPI-CUDA-AFQMC-Real,
+            TeslaV100-Clang16-MPI-CUDA-AFQMC-Offload-Real-Mixed, # auxiliary field, offload
```
Comment (ye-luo) [inline .github/workflows/ci-github-actions-self-hosted.yaml:118]: TeslaV100->NV_V100.

File: .github/workflows/ci-github-actions-self-hosted.yaml [hunk 2]
```diff
@@ -115,13 +115,13 @@ jobs:
...
+            TeslaV100-Clang16-MPI-CUDA-AFQMC-Offload-Real,
+            TeslaV100-Clang16-MPI-CUDA-AFQMC-Offload-Complex-Mixed,
+            TeslaV100-Clang16-MPI-CUDA-AFQMC-Offload-Complex,
+            TeslaV100-ICX23-MPI-CUDA-Real-Mixed,
```
Comment (ye-luo) [inline .github/workflows/ci-github-actions-self-hosted.yaml:122]: ICX23->OneAPI23.x
Please use the exact x.

File: .github/workflows/ci-github-actions-self-hosted.yaml (hunk 1 above)
Comment (prckent) [reply .github/workflows/ci-github-actions-self-hosted.yaml:118]: Yes, Tesla is redundant here. Can we use hyphens - everywhere or do you have a reason for underscores?

File: .github/workflows/ci-github-actions-self-hosted.yaml (hunk 2 above)
Comment (prckent) [reply .github/workflows/ci-github-actions-self-hosted.yaml:122]: Perhaps "OneAPI23"? We'll presumably update to the latest minor version shortly after release. If the minor version .x is in the name this creates administrative work updating the merge requirements on GitHub.

**Comment by williamfgc** [comment]
> What's the difference between run_step_ornl-sulfur-1.sh and run_step_ornl-sulfur-2.sh ? It looks like *-1.sh is for the CPU build, and *-2.sh is for the offload build?

That's correct, it was part of a previous refactoring to run CPU and GPU job groups concurrently to minimize CI times.

File: .github/workflows/ci-github-actions-self-hosted.yaml (hunk 1 above)
Comment (prckent) [reply .github/workflows/ci-github-actions-self-hosted.yaml:118]: V100 should be fine on its own if we are happy with RadeonVII

Comment (williamfgc) [reply .github/workflows/ci-github-actions-self-hosted.yaml:118]: Just V100? CUDA is in the name.

File: .github/workflows/ci-github-actions-self-hosted.yaml (hunk 2 above)
Comment (williamfgc) [reply .github/workflows/ci-github-actions-self-hosted.yaml:122]: OneAPI23 comes with classic icc and OneAPI icx, I used icx as a way to distinguish. Moving forward just OneAPI should be enough. Thoughts?

**Review by prckent — APPROVED** [review]

File: .github/workflows/ci-github-actions-self-hosted.yaml (hunk 1 above)
Comment (ye-luo) [reply .github/workflows/ci-github-actions-self-hosted.yaml]: Just V100 is fine. I also use R7 instead of RadeonVII.

**Pull request merged**